gcp_variant_transforms/testing/integration/run_tests_common.py

```diff
@@ -41,42 +41,57 @@ class TestRunner(object):
   """Runs the tests using pipelines API."""
 
   def __init__(self, tests):
-    # type: (List[Union[PreprocessorTestCase, VcfToBQTestCase]]) -> None
+    # type: (List[List[Union[PreprocessorTestCase, VcfToBQTestCase]]]) -> None
     self._tests = tests
     self._service = discovery.build(
         'genomics',
         'v1alpha2',
         credentials=GoogleCredentials.get_application_default())
-    self._operation_names = []  # type: List[str]
-    self._responses = []  # List[Dict]
+    self._operation_names_to_tests = {}  # type: Dict
```

Comment:
As discussed offline: Please make this dict be operation_names_to_test_states, where the key is the operation_name and the value is a namedtuple TestCaseState(RunningTest, RemainingTests). This essentially makes a 'queue' for you and you can remove a test from the RemainingTests and run it at each step.